larray/viewer.py

```diff
@@ -2332,34 +2358,50 @@ def new(self):
     def _open_file(self, filepath):
         # XXX : clear console history ?
         self._clear_arrays()
-        session = la.Session(filepath)
+        session = la.Session()
+        if '.csv' in filepath:
+            filepath = [filepath]
+        if isinstance(filepath, (list, tuple)):
+            directory = os.path.dirname(filepath[0])
+            names = [os.path.splitext(os.path.basename(fp))[0] for fp in filepath]
+            session.load(directory, names)
+            for name in names:
+                self.set_current_file('{}/{}.csv'.format(directory, name))
```

Comment:
This smells like a small refactor is necessary: move most of the functionality of set_current_file to a method called something like "update_recent_files". Then make set_current_file call that method and call that method here instead of using set_current_file.